server/routes/api/auth.js

```diff
@@ -0,0 +1,128 @@
+var router = require('express').Router(),
+    userModel = require('../../db/model/User.js'),
+    authMiddleware = require('../../middleware/auth.js');
+
+// Disable all non-post methods for /v1/auth
+router.use(function(req, res, next) {
+    if (req.method !== 'POST') {
+        res.status(405).send({
+            status: false,
+            message: "Method not allowed"
+        });
+    }
+    return next();
+});
+
+// Handles /v1/auth/login
+router.post('/login', function(req, res) {
+    // Check if the login request contains the email and password
+    if (req.body.email && req.body.password) {
+        // Lookup users with the email provided in the post body
+        userModel.find().byEmail(req.body.email).exec(function(err, model) {
+            // If there are no errors and the model is valid, check the password
+            // and allow the user in. If not, shutdown the request
+            if (!err) {
+                if (model) {
+                    model.checkPassword(req.body.password, function(checkErr, checkRes) {
+                        if (checkRes) {
+                            req.session.loggedIn = true;
+                            res.send({
+                                status: true,
+                                message: "Successfully authenticated",
+                                token: model.generateNewToken()
+                            });
+                        } else {
+                            res.status(401).send({
+                                status: false,
+                                message: "Unauthorized, email and password combination is invalid."
+                            });
+                        }
+                    });
+                } else {
+                    res.status(401).send({
+                        status: false,
+                        message: "Unauthorized, email and password combination is invalid."
+                    });
+                }
+            } else {
+                console.error(err);
+                res.status(500).send({
+                    status: false,
+                    message: "There was an error with the request, please try again later"
+                });
+            }
+        });
+    } else {
+        res.status(401).send({
+            status: false,
+            message: "Unauthorized, email and password combination is invalid."
```

Comment:
I feel like we should differentiate this response from bad email/password response on line 37